model/testing/src/icon4py/model/testing/grid_utils.py

```diff
@@ -84,17 +85,24 @@ def _file_name(grid_file: str):
 
 
 def resolve_full_grid_file_name(grid_file_str: str) -> pathlib.Path:
-    return dt_utils.GRIDS_PATH.joinpath(grid_file_str, _file_name(grid_file_str))
+    return definitions.grids_path().joinpath(grid_file_str, _file_name(grid_file_str))
 
 
 def _download_grid_file(file_path: str) -> pathlib.Path:
     full_name = resolve_full_grid_file_name(file_path)
-    if not full_name.exists():
-        data_handling.download_and_extract(
-            dt_utils.GRID_URIS[file_path],
-            full_name.parent,
-            full_name.parent,
-        )
+    grid_directory = full_name.parent
+    grid_directory.mkdir(parents=True, exist_ok=True)
+    with locking.lock(grid_directory):
+        if not full_name.exists():
+            if config.ENABLE_GRID_DOWNLOAD:
```

Comment:
I added this to get a nicer error in case downloading is not expected to happen.